Fix bug: populate cache deps [skip ci]

YanxuanLiu opened the pull request:
<!--

Thank you for contributing to RAPIDS Accelerator for Apache Spark!

Here are some guidelines to help the review process go smoothly.

1. Please write a description in this text box of the changes that are being
   made.

2. Please ensure that you have written units tests for the changes made/features
   added.

3. If you are closing an issue please use one of the automatic closing words as
   noted here: https://help.github.com/articles/closing-issues-using-keywords/

4. If your pull request is not ready for review but you want to make use of the
   continuous integration testing facilities please label it with `[WIP]`.

5. If your pull request is ready to be reviewed without requiring additional
   work on top of it, then remove the `[WIP]` label (if present).

6. Once all work has been done and review has taken place please do not add
   features or make changes out of the scope of those requested by the reviewer
   (doing this just add delays as already reviewed code ends up having to be
   re-reviewed/it is hard to tell what is new etc!). Further, please avoid
   rebasing your branch during the review process, as this causes the context
   of any comments made by reviewers to be lost. If conflicts occur during
   review then they should be resolved by merging into the branch used for
   making the pull request.

Many thanks in advance for your cooperation!

-->

fix https://github.com/NVIDIA/spark-rapids/issues/11796

Corrected the argument of `build/get_buildvers.py` to fix bug of getting build versions.

pxLi commented:
I think the original ticket https://github.com/NVIDIA/spark-rapids/issues/11796 is also asking for `fix both error masking and the underlying issues`

so we should correctly error out for pipeline usage, please try with `set -o pipefail`

YanxuanLiu commented:
> I think the original ticket #11796 is also asking for `fix both error masking and the underlying issues`
> 
> so we should correctly error out for pipe usage, please try with `set -o pipefail`

Updated the PR:

1. added `set -o pipefail` to catch errors in pipe
2. split commands to ensure `set -e` can work as expected.


Test build: https://github.com/YanxuanLiu/spark-rapids/actions/runs/12146768557/job/33871438631

gerashegalov commented on .github/workflows/mvn-verify-check/populate-daily-cache.sh:
```diff
@@ -14,27 +14,30 @@
 
-set -x
-max_retry=3; delay=30; i=1
+set -e
```
would be good to standardize in the future if we enable debug output by default.

YanxuanLiu replied:
Tested in [FAILED](https://github.com/YanxuanLiu/spark-rapids/actions/runs/12151847128/job/33887174202), the workflow failed with the bug.

Yes, for this script we disable debug output for clear log.

gerashegalov reviewed (approved):
LGTM

NvTimLiu reviewed (approved):
LGTM

YanxuanLiu commented:
build
